receiver/receivercreator/factory.go

```diff
@@ -23,21 +23,23 @@ import (
 	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/sharedcomponent"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/receivercreator/internal/metadata"
 )
 
 // This file implements factory for receiver_creator. A receiver_creator can create other receivers at runtime.
 
-const (
-	typeStr   = "receiver_creator"
-	stability = component.StabilityLevelBeta
-)
+var receivers = sharedcomponent.NewSharedComponents()
 
 // NewFactory creates a factory for receiver creator.
 func NewFactory() receiver.Factory {
 	return receiver.NewFactory(
-		typeStr,
+		metadata.Type,
 		createDefaultConfig,
-		receiver.WithMetrics(createMetricsReceiver, stability))
+		receiver.WithLogs(createLogsReceiver, metadata.Stability),
+		receiver.WithMetrics(createMetricsReceiver, metadata.Stability),
+		receiver.WithTraces(createTracesReceiver, metadata.Stability),
```

Comment:
Once `make generate` is run w/ the latest main these will need updating as well:
```suggestion
		receiver.WithLogs(createLogsReceiver, metadata.LogsStability),
		receiver.WithMetrics(createMetricsReceiver, metadata.MetricsStability),
		receiver.WithTraces(createTracesReceiver, metadata.TracesStability),
```